src/accessories/CampaignImporterExporter.ttslua

```diff
@@ -224,20 +255,38 @@ function createCampaignToken(_, playerColor, _)
     return
   end
 
+  -- clean up chaos tokens
+  blessCurseApi.removeAll(playerColor)
+  chaosBagApi.releaseAllSealedTokens(playerColor)
+
   local campaignData = {
     box = campaignBox.getGUID(),
-    log = campaignLog.getData(),
+    log = campaignLog.getPosition(),
     bag = chaosBagApi.getChaosBagState(),
     trauma = traumaValues,
     decks = deckImporterApi.getUiState(),
     clueCount = playAreaApi.getInvestigatorCount(),
-    guide = campaignGuide.Book.getPage(),
+    playarea = playAreaApi.getSurface(),
     options = optionPanelApi.getOptions(),
-    playmat = playAreaApi.getSurface()
+    guide = campaignGuide.Book.getPage(),
+    additionalIndex = additionalIndex.getPosition()
   }
   campaignTokenData.GMNotes = JSON.encode(campaignData)
-  campaignTokenData.Nickname = os.date("%b %d ") .. campaignBox.getName() .. " Save"
-  spawnObjectData({ data = campaignTokenData })
+  campaignTokenData.Nickname = campaignBox.getName() .. os.date(" %b %d") .. " Save"
+  spawnObjectData({ data = campaignTokenData, callback_function = finishTokenSpawning })
+end
+
+function finishTokenSpawning(obj)
+  local logData = campaignLog.getData()
+  local indexData = additionalIndex.getData()
+  campaignLog.setLock(false)
+  additionalIndex.setLock(false)
+  checkWarning = false
+  obj.putObject(additionalIndex)
+  obj.putObject(campaignLog)
+  checkWarning = true
+  spawnObjectData({data = indexData})
+  spawnObjectData({data = logData})
```

Comment:
Instead of removing the objects and then respawning them, it would be better to use `getData()` and then add that data to the `ContainedObjects` of the coin